src/Intl/Transliterator/Transliterator.php

```diff
@@ -0,0 +1,761 @@
+     */
+    private static function charsArrayWithSingleLanguageValues()
+    {
+        // init
+        static $CHARS_ARRAY = null;
+
+        if (isset($CHARS_ARRAY)) {
+            return $CHARS_ARRAY;
+        }
+
+        self::prepareAsciiMaps();
+
+        /* @noinspection AlterInForeachInspection */
+        /* @psalm-suppress PossiblyNullIterator - we use the prepare* methods here, so we don't get NULL here */
+        foreach (self::$ASCII_MAPS as &$map) {
+            $CHARS_ARRAY[] = $map;
+        }
+
+        $CHARS_ARRAY = \call_user_func_array('array_merge', $CHARS_ARRAY + array());
+
+        $CHARS_ARRAY = array(
+            'orig' => array_keys($CHARS_ARRAY),
+            'replace' => array_values($CHARS_ARRAY),
+        );
+
+        return $CHARS_ARRAY;
+    }
+
+    private static function clean_id($s)
+    {
+        return rtrim($s, ';');
+    }
+
+    public static function create($id, $direction = \Transliterator::FORWARD)
+    {
+        if (
+            false !== strpos($id, '<')
+            ||
+            false !== strpos($id, '>')
+            ||
+            false !== strpos($id, '::')
+        ) {
+            return null;
+        }
+
+        $transliterator = new self();
+
+        $transliterator->id_intern = self::clean_id($id);
+
+        if (\Transliterator::FORWARD !== $direction && null !== $direction) {
+            throw new \DomainException(sprintf('The PHP intl extension is required for using "Transliterator->direction".'));
+        }
+
+        return $transliterator;
+    }
+
+    public static function createFromRules($rules, $direction = \Transliterator::FORWARD)
+    {
+        if (
+            false === strpos($rules, '<')
+            &&
+            false === strpos($rules, '>')
+            &&
+            false === strpos($rules, '::')
+        ) {
+            return null;
+        }
+
+        $transliterator = new self();
+
+        $transliterator->id_intern = self::clean_id($rules);
+
+        if (\Transliterator::FORWARD !== $direction && null !== $direction) {
+            throw new \DomainException(sprintf('The PHP intl extension is required for using "Transliterator->direction".'));
+        }
+
+        return $transliterator;
+    }
+
+    public static function createInverse()
+    {
+        throw new \DomainException(sprintf('The PHP intl extension is required for using "Transliterator::createInverse".'));
+    }
+
+    /**
+     * Get data from "/data/*.php".
+     *
+     * @param string $file
+     *
+     * @return array
+     */
+    private static function getData($file)
+    {
+        /* @noinspection PhpIncludeInspection */
+        /* @noinspection UsingInclusionReturnValueInspection */
+        /* @psalm-suppress UnresolvableInclude */
+        return include __DIR__.'/Resources/unidata/'.$file.'.php';
+    }
+
+    /**
+     * Get data from "/data/*.php".
+     *
+     * @param string $file
+     *
+     * @return array|false
```

Comment:
`array|null` would be better as signature.

Reply:
now we return always an array here